diff -ruN a/libmp3lame/bitstream.c b/libmp3lame/bitstream.c
--- a/libmp3lame/bitstream.c	2017-09-06 21:33:35
+++ b/libmp3lame/bitstream.c	2026-09-20 13:36:01
@@ -466,11 +466,6 @@
         }
     }
 
-    if (cfg->error_protection) {
-        /* (jo) error_protection: add crc16 information to header */
-        CRC_writeheader(gfc, esv->header[esv->h_ptr].buf);
-    }
-
     {
         int const old = esv->h_ptr;
         assert(esv->header[old].ptr == cfg->sideinfo_len * 8);
diff -ruN a/libmp3lame/presets.c b/libmp3lame/presets.c
--- a/libmp3lame/presets.c	2012-10-06 13:44:56
+++ b/libmp3lame/presets.c	2026-09-20 13:36:01
@@ -359,42 +359,6 @@
     }
 
     gfp->preset = preset;
-    {
-        switch (preset) {
-        case V9:
-            apply_vbr_preset(gfp, 9, enforce);
-            return preset;
-        case V8:
-            apply_vbr_preset(gfp, 8, enforce);
-            return preset;
-        case V7:
-            apply_vbr_preset(gfp, 7, enforce);
-            return preset;
-        case V6:
-            apply_vbr_preset(gfp, 6, enforce);
-            return preset;
-        case V5:
-            apply_vbr_preset(gfp, 5, enforce);
-            return preset;
-        case V4:
-            apply_vbr_preset(gfp, 4, enforce);
-            return preset;
-        case V3:
-            apply_vbr_preset(gfp, 3, enforce);
-            return preset;
-        case V2:
-            apply_vbr_preset(gfp, 2, enforce);
-            return preset;
-        case V1:
-            apply_vbr_preset(gfp, 1, enforce);
-            return preset;
-        case V0:
-            apply_vbr_preset(gfp, 0, enforce);
-            return preset;
-        default:
-            break;
-        }
-    }
     if (8 <= preset && preset <= 320) {
         return apply_abr_preset(gfp, preset, enforce);
     }
diff -ruN a/libmp3lame/quantize.c b/libmp3lame/quantize.c
--- a/libmp3lame/quantize.c	2017-08-15 15:40:45
+++ b/libmp3lame/quantize.c	2026-09-20 13:36:01
@@ -431,101 +431,8 @@
 
 
 
-/************************************************************************
- *
- *      trancate_smallspectrums()
- *
- *  Takehiro TOMINAGA 2002-07-21
- *
- *  trancate smaller nubmers into 0 as long as the noise threshold is allowed.
- *
- ************************************************************************/
-static int
-floatcompare(const void *v1, const void *v2)
-{
-    const FLOAT *const a = v1, *const b = v2;
-    if (*a > *b)
-        return 1;
-    if (*a < *b)
-        return -1;
-    return 0;
-}
 
-static void
-trancate_smallspectrums(lame_internal_flags const *gfc,
-                        gr_info * const gi, const FLOAT * const l3_xmin, FLOAT * const work)
-{
-    int     sfb, j, width;
-    FLOAT   distort[SFBMAX];
-    calc_noise_result dummy;
 
-    if ((!(gfc->sv_qnt.substep_shaping & 4) && gi->block_type == SHORT_TYPE)
-        || gfc->sv_qnt.substep_shaping & 0x80)
-        return;
-    (void) calc_noise(gi, l3_xmin, distort, &dummy, 0);
-    for (j = 0; j < 576; j++) {
-        FLOAT   xr = 0.0;
-        if (gi->l3_enc[j] != 0)
-            xr = fabs(gi->xr[j]);
-        work[j] = xr;
-    }
-
-    j = 0;
-    sfb = 8;
-    if (gi->block_type == SHORT_TYPE)
-        sfb = 6;
-    do {
-        FLOAT   allowedNoise, trancateThreshold;
-        int     nsame, start;
-
-        width = gi->width[sfb];
-        j += width;
-        if (distort[sfb] >= 1.0)
-            continue;
-
-        qsort(&work[j - width], width, sizeof(FLOAT), floatcompare);
-        if (EQ(work[j - 1], 0.0))
-            continue;   /* all zero sfb */
-
-        allowedNoise = (1.0 - distort[sfb]) * l3_xmin[sfb];
-        trancateThreshold = 0.0;
-        start = 0;
-        do {
-            FLOAT   noise;
-            for (nsame = 1; start + nsame < width; nsame++)
-                if (NEQ(work[start + j - width], work[start + j + nsame - width]))
-                    break;
-
-            noise = work[start + j - width] * work[start + j - width] * nsame;
-            if (allowedNoise < noise) {
-                if (start != 0)
-                    trancateThreshold = work[start + j - width - 1];
-                break;
-            }
-            allowedNoise -= noise;
-            start += nsame;
-        } while (start < width);
-        if (EQ(trancateThreshold, 0.0))
-            continue;
-
-/*      printf("%e %e %e\n", */
-/*             trancateThreshold/l3_xmin[sfb], */
-/*             trancateThreshold/(l3_xmin[sfb]*start), */
-/*             trancateThreshold/(l3_xmin[sfb]*(start+width)) */
-/*          ); */
-/*      if (trancateThreshold > 1000*l3_xmin[sfb]*start) */
-/*          trancateThreshold = 1000*l3_xmin[sfb]*start; */
-
-        do {
-            if (fabs(gi->xr[j - width]) <= trancateThreshold)
-                gi->l3_enc[j - width] = 0;
-        } while (--width > 0);
-    } while (++sfb < gi->psymax);
-
-    gi->part2_3_length = noquant_count_bits(gfc, gi, 0);
-}
-
-
 /*************************************************************************
  *
  *      loop_break()
@@ -1188,10 +1095,6 @@
     if (cfg->vbr == vbr_rh || cfg->vbr == vbr_mtrh || cfg->vbr == vbr_mt)
         /* restore for reuse on next try */
         memcpy(xrpow, save_xrpow, sizeof(FLOAT) * 576);
-    /*  do the 'substep shaping'
-     */
-    else if (gfc->sv_qnt.substep_shaping & 1)
-        trancate_smallspectrums(gfc, cod_info, l3_xmin, xrpow);
 
     return best_noise_info.over_count;
 }
@@ -1533,12 +1436,6 @@
 
                 VBR_encode_granule(gfc, cod_info, l3_xmin[gr][ch], xrpow,
                                    ch, min_bits[gr][ch], max_bits[gr][ch]);
-
-                /*  do the 'substep shaping'
-                 */
-                if (gfc->sv_qnt.substep_shaping & 1) {
-                    trancate_smallspectrums(gfc, &l3_side->tt[gr][ch], l3_xmin[gr][ch], xrpow);
-                }
 
                 ret = cod_info->part2_3_length + cod_info->part2_length;
                 used_bits += ret;
